### ISSUE TITLE
IE11 performance: Make css custom property processing optional

### ISSUE THREAD

**Issue by cary-smith** [ionic-team/stencil#2059]
**Stencil version:**

```
 @stencil/core@1.8.x
```

**I'm submitting a:**

[ ] bug report
[x] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**

When using our library in an Angular application that has imported the angular-mdc-web full theme (I know this is not advisable, but some folks just grab the whole bag), if they are using our Stencil components in applications intended for consumption on IE11 it can take 3 minutes to load. When digging through the slowdown it appears that Stencil is rewriting a ton of css unnecessarily. 

Angular MDC delivers its css for the "Ancient One" as well as Evergreen browsers:
```css
color: blue; /* IE fail compatible */
color: var(--mdc-some-color, blue); /* Pretty much every functional browser */
```
Stencil's css shim is parsing and rewriting this for IE11 so we get:
```css
color: blue;
color: blue;
```
Super-cool, but it is currently causing a slowdown in Angular which makes our Stencil-built library seem incompatible with libraries that use custom properties. 

**Expected behavior:**

I thought about this but have only come up with a couple of ideas:

1. Allow us to disable this behavior. IE11 is the bane of the web world and I would much rather provide a functional application that only works with the fallback css because IE11 users should not get the cool toys especially if it makes them late for school. Actually...they just should not get any cool toys. 😞 
1. Do not replace css rules that are immediately preceded by a fallback that does not have `var()` in it. I know this would be heavier initial work, but would reduce how much the parser would be replacing with duplication. 
1. Make the functionality a plugin that can be used for those that need it like `@stencil/sass`.

**Steps to reproduce:**

I am working on a test case but cannot find the right combination of vectors to recreate this in isolation. I will update the ticket when I can. 😕 

**Related code:**

```tsx
// insert any relevant code here
```

**Other information:**

If there is an `Off` switch and I just could not find it do to ignorance, please let me know. 

As a note: The full usage of angular-mdc pulls in close to 800+ custom props that have to be checked and "re-written". Each parse of the css header (when it is does the text replace) makes IE11 choke for a bit longer. For whatever (still unknown) reason, this listing is being parsed multiple times during the initial load.

**Comment by cary-smith** [comment]
In versions of Stencil before the css prop was being auto-replaced, we used a rather simple sass method to isolate all IE11-specific rules: 
```scss
@mixin ie11Only {
    @media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
        @content;
    }
}
```
Sometimes we did it for performance reasons, sometimes for management of The Terrible Browser via fallbacks. I know you have added in slotting functionality (which I absolutely appreciate) for legacy browsers but I would rather control the CSS failures of IE11 and just provide a 'lesser' experience to IE11 users as long as it is still somewhat performant.

**Comment by cary-smith** [comment]
Looks like this will be addressed in 1.9 when it goes live a a global config!:

https://github.com/ionic-team/stencil/commit/b3b69933528b7106ff69720797ecaabec74e4546

**Comment by cary-smith** [comment]
@adamdbradley - Thanks for making this optional! Everything is running much smoother now.
